zrml/orderbook/src/migrations.rs

```diff
@@ -0,0 +1,336 @@
+                    old_order.base_asset,
+                    old_order.base_asset_amount,
+                ),
+            };
+
+            let new_order = Order {
+                market_id: old_order.market_id,
+                maker: old_order.maker,
+                maker_asset,
+                maker_amount,
+                taker_asset,
+                taker_amount,
+            };
+
+            Some(new_order)
+        });
+        log::info!("TranslateOrderStructure: Upgraded {} orders.", translated);
+        total_weight =
+            total_weight.saturating_add(T::DbWeight::get().reads_writes(translated, translated));
+
+        StorageVersion::new(ORDER_BOOK_NEXT_STORAGE_VERSION).put::<OrderbookPallet<T>>();
+        total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        log::info!("TranslateOrderStructure: Done!");
+        total_weight
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+        use frame_support::pallet_prelude::Twox64Concat;
+
+        let old_orders =
+            storage_key_iter::<OrderId, OldOrderOf<T>, Twox64Concat>(ORDER_BOOK, ORDERS)
+                .collect::<BTreeMap<_, _>>();
+
+        let orders = Orders::<T>::iter_keys().count() as u32;
+        let decodable_orders = Orders::<T>::iter_values().count() as u32;
+        if orders == decodable_orders {
+            log::info!("All orders could successfully be decoded, order_count: {}.", orders);
+        } else {
+            log::error!(
+                "Can only decode {} of {} orders - others will be dropped",
+                decodable_orders,
+                orders
+            );
+        }
+
+        Ok(old_orders.encode())
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade(previous_state: Vec<u8>) -> Result<(), &'static str> {
+        use orml_traits::NamedMultiReservableCurrency;
+        use sp_runtime::traits::Zero;
+        use zeitgeist_primitives::traits::MarketCommonsPalletApi;
+
+        let old_orders: BTreeMap<OrderId, OldOrderOf<T>> = Decode::decode(&mut &previous_state[..])
+            .expect("Failed to decode state: Invalid state");
+        let mut new_order_count = 0usize;
+        assert_eq!(old_orders.len(), new_order_count);
+        for (order_id, new_order) in crate::Orders::<T>::iter() {
+            let old_order =
+                old_orders.get(&order_id).expect(&format!("Order {:?} not found", order_id)[..]);
+            // assert old fields
+            assert_eq!(old_order.market_id, new_order.market_id);
+            assert_eq!(old_order.maker, new_order.maker);
+            // assert new fields
+            let reserved = T::AssetManager::reserved_balance_named(
+                &OrderbookPallet::<T>::reserve_id(),
+                new_order.maker_asset,
+                &new_order.maker,
+            );
+            assert!(reserved > BalanceOf::<T>::zero());
```

Comment:
```suggestion
```

Reply:
This is covered by the next `assert_eq`

Reply:
186b1d2ac14187f85bb95f25c37019c501fb6122